src/Dscp/Workers/Worker.hs

```diff
@@ -7,19 +7,73 @@ module Dscp.Workers.Worker
     ) where
 
 import Control.Concurrent (threadDelay)
-import Fmt ((+|), (||+))
+import Fmt ((+||), (||+))
 import Loot.Log (logError, logInfo)
 import Loot.Network.Class (ClientEnv)
 import Loot.Network.ZMQ (ZmqTcp)
+--import qualified Snowdrop.Model.Block as SD
+--import qualified Snowdrop.Model.Execution as SD
+--import qualified Snowdrop.Model.State.Core as SD
+--import qualified Snowdrop.Util as SD
 
+import Dscp.Core
 import Dscp.Network.Messages (PingBlk (..), PingTx (..), PongBlk (..), PongTx (..))
 import Dscp.Network.Wrapped (Worker (..), cliRecvResp, cliSend, msgType)
+import Dscp.Slotting
 import Dscp.Witness.Launcher (WitnessWorkMode)
+import Dscp.Witness.Mempool
 
-witnessWorkers :: WitnessWorkMode m => [Worker ZmqTcp m]
-witnessWorkers = [witnessTxWorker, witnessBlkWorker]
 
-witnessTxWorker :: forall m. WitnessWorkMode m => Worker ZmqTcp m
+witnessWorkers :: WitnessWorkMode ctx m => [Worker m]
+witnessWorkers = [blockIssuingWorker, witnessTxWorker, witnessBlkWorker]
+
+----------------------------------------------------------------------------
+-- Block creation
+----------------------------------------------------------------------------
+
+_createPayload :: MonadIO m => MempoolVar -> m BlockBody
+_createPayload v = BlockBody <$> swapTxsMempool v
+
+_createBlock :: WitnessWorkMode ctx m => m Block
+_createBlock = error "TBD DSCP-133"
```

Comment:
This is not supposed to be implemented here? When yes, please create a separate task for it.

Reply:
I'll continue DSCP-133 right after this PR will be merged. I've changed a lot of logic so I want others to have it already.